code-collab-app/src/uploadpopup.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';  
+import './App.css';
+import { Button, ButtonGroup } from 'react-bootstrap';
+import { inject, observer } from 'mobx-react';
+
+class Popup extends React.Component {
+  constructor(props){
+    super(props);
+    
+    this.file_reader = new FileReader();
+    this.file_name = "Choose a file..."
+  }
+
+  onChangeHandler = event => {
+    this.setState({
+      selectedFile: event.target.files[0]
+    });
+    this.file_name = event.target.files[0].name;
+  };
+
+  handleFileRead = (e) => {
+    let divider = "--------------------------------------------------------------";
```

Comment:
Did you test this with the download button? Just wondering if we'll need `\n` on each end of the divider or something to handle the extra newlines inserted for the downloaded file

Reply:
Good catch! I will fix this in the next PR